docs/static/doc-for-plugin.asciidoc

```diff
@@ -0,0 +1,184 @@
+-----
+
+
+If you want your link to display as {logstash-ref}/getting-started-with-logstash.html[Getting Started with Logstash], use this format:
+[source,txt]
+-----
+{logstash-ref}/getting-started-with-logstash.html[Getting Started with Logstash]
+-----
+
+
+===== Link to data type descriptions
+
+We make an exception for links that point to data type descriptions,
+such as `<<boolean,boolean>>`, because they are used so frequently. 
+We have a cleanup step in the conversion script that converts the links to the
+correct syntax. 
+
+
+[[format-code]]
+==== Code samples
+
+We all love code samples. Asciidoc supports code blocks and config examples.
+To include Ruby code, use the asciidoc `[source,ruby]` directive.
+
+Note that the hashmarks (#) are present to make the example render correctly.
+Don't include the hashmarks in your asciidoc file.
+
+[source,txt]
+[subs="attributes"]
+----------------------------------
+# [source,ruby]
+# -----
+# match => {
+#  "field1" => "value1"
+#  "field2" => "value2"
+#  ...
+# }
+# -----
+----------------------------------
+
+The sample above (with hashmarks removed) renders in the documentation like this:
+[source,ruby]
+----------------------------------
+match => {
+  "field1" => "value1"
+  "field2" => "value2"
+  ...
+}
+----------------------------------
+
+==== Where's my doc?
+
+Plugin documentation goes through several steps before it gets published in the 
+{lsplugindocs}[Logstash Versioned Plugin Reference] and the {logstash-ref}/index.html[Logstash Reference].
+
+Here's an overview of the workflow:
+
+* Be sure that you have signed the CLI and have all necessary approvals and sign offs.
+* Merge the pull request for your plugin (including the `index.asciidoc` file, the `changelog.md` file, and the gemspec).
+* Wait for the continuous integration build to complete successfully.
+* Publish the plugin to https://rubygems.org.
+* A script detects the new or changed version, and picks up the `index.asciidoc` file for inclusion in the doc build.
+* The documentation for your new plugin is published in the {lsplugindocs}[Logstash Versioned Plugin Reference].
+
+We're not done yet. 
+
+* For each release, we package the new and changed documentation files into a pull request to add or update content.
+(We sometimes package plugin docs between releases if we make significant changes to plugin documentation or add a new plugin.)
+* The script detects the new or changed version, and picks up the `index.asciidoc` file for inclusion in the doc build.
+* We create a pull request, and merge the new and changed content into the appropriate version branches.
+* For a new plugin, we add a link to the list of plugins in the {logstash-ref}/index.html[Logstash Reference].
+* The documentation for your new (or changed) plugin is published in the {logstash-ref}/index.html[Logstash Reference].
+
+===== Documentation or plugin updates
+
+When you make updates to your plugin or the documentation, consider
```

Comment:
ah here you mention it, but I'm not sure it's clear that you *must* do this to get the content published.